### ISSUE TITLE
Balance error

### ISSUE THREAD

**Issue by wd021** [iron-fish/ironfish#2553]
### What happened?

tldr - tried to send my entire balance **0.48334460**, didn't work. did a full rescan the same thing. So then i tried sending **0.38334940** and it worked. The new balance showed **0.09999520** and sent it all in a transaction that worked. The amounts add up to **0.38334940 + 0.09999520 = 0.4833446**. The original balance showing **0.48334460** was off by **0.00000014** (i did a full rescan to make sure the original balance was correct). 

Here's the full flow 👇 

1) get account balance
```
Account: d91fd4c208d66cc147d2f8e2554e597b1dd81aa851d18093bd6a9c25a0c035fc2e603b847129b11d72e203
Balance: $IRON 0.48334460
```

2) I try to send it all and get insufficient balance error. Double checked to make sure the amount adds up, ran accounts:balance --explain to see status and did a full rescan, but always kept getting the same error.
```
yarn start accounts:pay -a 0.483343900 -o 0.00000560 -t 421df52c7b26a78e520ad461bea95068a3de22fad60a5ea3b3122aec6736f8baa053a542f557ed633e03ef

You are about to send:
$IRON 0.48334390 plus a transaction fee of $IRON 0.00000560 to 421df52c7b26a78e520ad461bea95068a3de22fad60a5ea3b3122aec6736f8baa053a542f557ed633e03ef from the account d91fd4c208d66cc147d2f8e2554e597b1dd81aa851d18093bd6a9c25a0c035fc2e603b847129b11d72e203

* This action is NOT reversible *

Do you confirm (Y/N)?: y
Creating the transaction: [████████████████████████████████████████] 100% | ETA: 0s
An error occurred while sending the transaction.
 ›   Error: Request failed (400) insufficient-balance: Your balance is too low. Add funds to your account first
```

3) i reduce the amount to send so its not the full balance and it goes through 
```
You are about to send:
$IRON 0.38334390 plus a transaction fee of $IRON 0.00000550 to 421df52c7b26a78e520ad461bea95068a3de22fad60a5ea3b3122aec6736f8baa053a542f557ed633e03ef from the account d91fd4c208d66cc147d2f8e2554e597b1dd81aa851d18093bd6a9c25a0c035fc2e603b847129b11d72e203


Do you confirm (Y/N)?: y
Creating the transaction: [████████████████████████████████████████] 100% | ETA: 0s

Sending $IRON 0.38334390 to 421df52c7b26a78e520ad461bea95068a3de22fad60a5ea3b3122aec6736f8baa053a542f557ed633e03ef from d91fd4c208d66cc147d2f8e2554e597b1dd81aa851d18093bd6a9c25a0c035fc2e603b847129b11d72e203
Transaction Hash: e1974fc6590d20b77a953e54b89a6ceb0619b220eb16bdca28903889e0f0cc51
Transaction fee: $IRON 0.00000550

Find the transaction on https://explorer.ironfish.network/transaction/e1974fc6590d20b77a953e54b89a6ceb0619b220eb16bdca28903889e0f0cc51 (it can take a few minutes before the transaction appears in the Explorer)
```

4) check new account balance
```
Account: d91fd4c208d66cc147d2f8e2554e597b1dd81aa851d18093bd6a9c25a0c035fc2e603b847129b11d72e203

Your balance is made of notes on the chain that are safe to spend
Balance: $IRON 0.00000000

0 notes worth $IRON 0.00000000 are on the chain within 2 blocks of the head
Unconfirmed: $IRON 0.00000000

1 notes worth $IRON 0.09999520 are waiting for miners to add them to the chain
Pending: $IRON 0.09999520
```

5) send the remainder (0.09999520) in a tx and it works
```
You are about to send:
$IRON 0.09999020 plus a transaction fee of $IRON 0.00000500 to 421df52c7b26a78e520ad461bea95068a3de22fad60a5ea3b3122aec6736f8baa053a542f557ed633e03ef from the account d91fd4c208d66cc147d2f8e2554e597b1dd81aa851d18093bd6a9c25a0c035fc2e603b847129b11d72e203

* This action is NOT reversible *

Do you confirm (Y/N)?: y
Creating the transaction: [████████████████████████████████████████] 100% | ETA: 0s

Sending $IRON 0.09999020 to 421df52c7b26a78e520ad461bea95068a3de22fad60a5ea3b3122aec6736f8baa053a542f557ed633e03ef from d91fd4c208d66cc147d2f8e2554e597b1dd81aa851d18093bd6a9c25a0c035fc2e603b847129b11d72e203
Transaction Hash: 53abf60c8046fcf5d1b9c174ae0bbcee0b4e1f81250502debf2d488cd0d4be2e
Transaction fee: $IRON 0.00000500

Find the transaction on https://explorer.ironfish.network/transaction/53abf60c8046fcf5d1b9c174ae0bbcee0b4e1f81250502debf2d488cd0d4be2e (it can take a few minutes before the transaction appears in the Explorer)
```

### Version

0.1.51

### Graffiti

wd021

**Comment by hughy** [comment]
This is very interesting 🤔 

Sending the transactions seems to have correctly set the balance to 0, but the amounts of the transactions don't add up to the amount of the balance.

**Comment by wd021** [comment]
yea, its a weird one. let me know if you need anything from my end to help debug this further.

**Comment by hughy** [comment]
I think what may have happened here is that calculating the inputs and outputs of the transactions from the command line information (outside the node) lost some precision and tricked us. The transaction note values in $ORE balance out when looking at the data in the database.

If we turn to Google to check the math we can see how we might lose precision: [in $IRON](https://www.google.com/search?q=0.38334940+%2B+0.09999520&ei=ZlJ2Y-SxD-XjkPIPs8OjuAE&ved=0ahUKEwjk_MKWw7X7AhXlMUQIHbPhCBcQ4dUDCA8&uact=5&oq=0.38334940+%2B+0.09999520&gs_lcp=Cgxnd3Mtd2l6LXNlcnAQA0oECEEYAUoECEYYAFAAWABg0hdoAnAAeACAAQCIAQCSAQCYAQCgAQLAAQE&sclient=gws-wiz-serp) but not [in $ORE](https://www.google.com/search?q=38334940+%2B+9999520&ei=zFN2Y93aE_vmkPIPovOP8A8&ved=0ahUKEwjd8KHBxLX7AhV7M0QIHaL5A_4Q4dUDCA8&uact=5&oq=38334940+%2B+9999520&gs_lcp=Cgxnd3Mtd2l6LXNlcnAQA0oECEEYAUoECEYYAFDsA1jsA2C2B2gBcAB4AIABSYgBSZIBATGYAQCgAQKgAQHAAQE&sclient=gws-wiz-serp)

Doing the same calculations with your original transaction inputs in $ORE: 48334390 + 560 = 48334950 > 48334460

So the original transaction was trying to send more than the full balance

**Comment by wd021** [comment]
@hughy i see, makes sense 👌
